source/postprocess/mass_error_statistics.cc

```diff
@@ -0,0 +1,225 @@
+                                global_norm_mass_error * year_in_seconds);
+
+          // also make sure that the other columns filled by the this object
+          // all show up with sufficient accuracy and in scientific notation
+          {
+            const char *columns[] = { "Integral mass error (kg/year)",
+                                      "Norm mass error (kg/year)"
+                                    };
+            for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
+              {
+                statistics.set_precision (columns[i], 8);
+                statistics.set_scientific (columns[i], true);
+              }
+          }
+        }
+      else
+        {
+          statistics.add_value ("Integral mass error (kg/s)",
+                                global_integral_mass_error);
+          statistics.add_value ("Norm mass error (kg/s)",
+                                global_norm_mass_error);
+
+          // also make sure that the other columns filled by the this object
+          // all show up with sufficient accuracy and in scientific notation
+          {
+            const char *columns[] = { "Integral mass error (kg/s)",
+                                      "Norm mass error (kg/s)"
+                                    };
+            for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
+              {
+                statistics.set_precision (columns[i], 8);
+                statistics.set_scientific (columns[i], true);
+              }
+          }
+        }
+
+      std::ostringstream output;
+      output.precision(3);
+      if (this->convert_output_to_years() == true)
+        output << global_integral_mass_error *year_in_seconds
+               << " kg/year, "
+               << global_norm_mass_error *year_in_seconds
+               << " kg/year";
+      else
+        output << global_integral_mass_error
+               << " kg/s, "
+               << global_norm_mass_error
+               << " kg/s";
+
+      return std::pair<std::string, std::string> ("Integral mass error, Norm mass error:",
+                                                  output.str());
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    ASPECT_REGISTER_POSTPROCESSOR(MassErrorStatistics,
+                                  "mass error statistics",
+                                  "A postprocessor that computes some statistics about the "
+                                  "error of the mass conservation equation. More precisely, it "
+                                  "computes the norm and the integral of the equation "
+                                  "$\\rho \\Nabla \\mathbf u + \\Nabla \\rho \\cdot \\mathbf u$. "
+                                  "Therefore, its results indicate how accurate the chosen approximation "
+                                  "for compressibility is for the given model setup.")
```

Comment:
Well, not quite. If you have inflow or outflow of the domain, then the result is *expected* to be nonzero!